### PR TITLE
add cleaner and change name

### PR DESCRIPTION
**Pull request by GhostFanta**

### CODE REVIEW

File: backend/src/main/resources/db/migration/V113__fix_sample_task_seeding.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 UPDATE template.sample_tasks
 SET due_time  = '21:43',
-    time_zone = 'America/New_York'
+    time_zone = 'America/Los_Angeles'
```
Comment (singerdmx) [inline backend/src/main/resources/db/migration/V113__fix_sample_task_seeding.sql:3]: Don't touch this file!

File: daemon/api/service/cleaner.go [hunk 2]
```diff
@@ -67,6 +67,13 @@ func (c *Cleaner) deleteByExpirationTimeBefore(tableName string) {
 	c.deleteFromTableByCond(expirationTimeBeforeCond, tableName)
 }
 
+func (c *Cleaner) deleteByAvailableBefore(tableName string) {
+	var staleTimeBeforeCond = db.Cond {
```
Comment (singerdmx) [inline daemon/api/service/cleaner.go:71]: How did you specify schema "template"?

File: daemon/clients/investment/ipo.go [hunk 3]
```diff
@@ -107,7 +107,7 @@ func (c *IPOClient) SendData() error {
 			UpdatedAt:       time.Now(),
 			Metadata:        "INVESTMENT_IPO_RECORD",
 			Content:         "",
-			Name:            target.Name,
+			Name:            fmt.Sprintf("%v goes public on %v", target.Name, availBefore),
```
Comment (singerdmx) [inline daemon/clients/investment/ipo.go:110]: Snowflake (SNOW) goes public on 2020-09-21

Are you missing the ticker here?

Comment (singerdmx) [reply daemon/clients/investment/ipo.go:110]: I mean SNOW

File: daemon/clients/investment/ipo.go [hunk 4]
```diff
@@ -107,7 +107,7 @@ func (c *IPOClient) SendData() error {
 			UpdatedAt:       time.Now(),
 			Metadata:        "INVESTMENT_IPO_RECORD",
 			Content:         "",
-			Name:            target.Name,
+			Name:            fmt.Sprintf("%v(%v) goes public on %v", target.Name, target.ticker, availBefore),
```
Comment (singerdmx) [inline daemon/clients/investment/ipo.go:110]: Add a space
%v (%v)

File: daemon/api/service/cleaner.go [hunk 5]
```diff
@@ -103,6 +110,7 @@ func (c *Cleaner) Clean(maxRetentionTimeInDays int) {
 	c.deleteByUpdatedAtBefore(t, "notifications")
 	c.deleteByUpdatedAtBefore(t, "auditables")
 	c.deleteByExpirationTimeBefore("public_project_items")
+	c.deleteByAvailableBefore("sample_task")
```
Comment (singerdmx) [inline daemon/api/service/cleaner.go:113]: change to template.sample_tasks

File: daemon/clients/investment/ipo.go [hunk 6]
```diff
@@ -107,7 +107,7 @@ func (c *IPOClient) SendData() error {
 			UpdatedAt:       time.Now(),
 			Metadata:        "INVESTMENT_IPO_RECORD",
 			Content:         "",
-			Name:            target.Name,
+			Name:            fmt.Sprintf("%v (%v) goes public on %v", target.Name, target.ticker, availBefore),
```
Comment (singerdmx) [inline daemon/clients/investment/ipo.go:110]: change availBefore to Date?
we need yyyy-mm-dd

Comment (singerdmx) [reply daemon/clients/investment/ipo.go:110]: availBefore is not what we want here
Change to string of yyyy-mm-dd

File: daemon/persistence/sampletask_dao.go [hunk 7]
```diff
@@ -34,36 +35,36 @@ func (s *SampleTaskDao) Upsert(t *SampleTask) {
...
-//		t.ReminderBeforeTask,
-//		t.DueDate,
-//		t.DueTime,
-//		t.UpdatedAt,
-//		t.AvailableBefore,
-//	)
-	//s.db.Exec(query)
-//}
\ No newline at end of file
+// func (s *SampleTaskDao) Upsert(t *SampleTask) {
```
Comment (singerdmx) [inline daemon/persistence/sampletask_dao.go:38]: why do we have upsert?

File: daemon/clients/investment/ipo.go [hunk 8]
```diff
@@ -107,14 +108,15 @@ func (c *IPOClient) SendData() error {
 			UpdatedAt:       time.Now(),
 			Metadata:        "INVESTMENT_IPO_RECORD",
 			Content:         "",
-			Name:            target.Name,
+			Name:            fmt.Sprintf("%v (%v) goes public on %v", target.Name, target.Ticker, t),
```
Comment (singerdmx) [inline daemon/clients/investment/ipo.go:111]: where is t defined?

Comment (stedong) [reply daemon/clients/investment/ipo.go:111]: t, _ := time.Parse(layoutISO, availBefore)

Comment (singerdmx) [reply daemon/clients/investment/ipo.go:111]: ok

File: backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql [hunk 9]
```diff
@@ -0,0 +1,2 @@
+ALTER SEQUENCE "template".sample_task_sequence RESTART WITH 8000;
```
Comment (singerdmx) [inline backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql:1]: remove

File: backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql [hunk 10]
```diff
@@ -0,0 +1,2 @@
...
+ALTER TABLE "template".sample_tasks ALTER COLUMN ID SET DEFAULT nextval('template.sample_task_sequence');
```
Comment (singerdmx) [inline backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql:2]: don't touch db schema in go!

File: backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql [hunk 11]
```diff
@@ -0,0 +1 @@
...
```
Comment (singerdmx) [inline backend/src/main/resources/db/migration/V126__set_sampletask_id_auto_increment.sql:1]: remove

**Pull request merged**